crates/ruff/src/rules/flake8_return/rules.rs

```diff
@@ -354,10 +369,75 @@ fn unnecessary_assign(checker: &mut Checker, stack: &Stack, expr: &Expr) {
             return;
         }
 
-        checker.diagnostics.push(Diagnostic::new(
-            UnnecessaryAssign,
-            Range::from_located(expr),
-        ));
+        let mut diagnostic = Diagnostic::new(UnnecessaryAssign, Range::from_located(expr));
+
+        if checker.patch(diagnostic.kind.rule()) {
+            if let Some(assign_expr) = stack.assign_values.get(id.as_str()) {
+                if let Some(assign_locations) = stack.assigns.get(id.as_str()) {
```

Comment:
Can we add some comments about what `assign_expr`, `assign_locations`, etc. represent here? I wrote some of this originally and even I can't remember what they mean and how they apply to this fix heh.

Reply:
Will do.